src/WrappedMToken.sol

```diff
@@ -443,7 +469,19 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
         emit Claimed(account_, claimRecipient_, yield_);
         emit Transfer(address(0), account_, yield_);
 
-        if (claimRecipient_ != account_) {
+        if (accountInfo_.hasEarnerDetails) {
+            (, uint16 feeRate_, address admin_) = IEarnerManager(earnerManager).getEarnerDetails(account_);
+
+            feeRate_ = feeRate_ > 10_000 ? 10_000 : feeRate_; // Ensure fee rate is capped at 100%.
+            uint240 fee_ = (feeRate_ * yield_) / 10_000;
+
+            if (fee_ != 0) {
+                _transfer(account_, admin_, fee_, currentIndex_);
```

Comment:
CEI is not in the right order 

```
yield_ -= fee_;
_transfer(account_, admin_, fee_, currentIndex_);
```

Reply:
While `yield_` is just a local variable, so it is not an effect, and no reentrancy can impact it, I'll swap the lines since there is no harm in treating setting local variable as "effects" for the purposes of CEI.